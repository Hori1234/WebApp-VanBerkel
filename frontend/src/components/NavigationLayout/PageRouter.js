import {Redirect, Route, Switch} from "react-router-dom";
import PrivateRoute from "../PrivateRoute";
import UploadButton from "../UploadButton/UploadButton";
import AccountManagementLayout from "../AccountManagerLayout/AccountManagementLayout";
import Logout from "../Logout/Logout";
import ManualPlanning from "../ManualPlanning/ManualPlanning";
import ViewPlanning from "../ViewPlanningPageLayout/ViewPlanning";
import DataVisualization from "../DataVisualization/DataVisualization";
import MonthlyDataAnalytics from "../MonthlyDataAnalytics/MonthlyDataAnalytics";
import Home from "../Home/Home";
import React from "react";

/**
 * Defines the routing for the pages in the NavigationLayout component.
 *
 * A private route only shows the component if the logged in user has the correct role.
 *
 * If a url is not matched with one of the routes, the user is redirected to the homepage.
 */
export default function PageRouter() {
    return (
        <Switch>
          <PrivateRoute
              path="/upload"
              requiredRoles={['planner', 'administrator']}>
            <UploadButton />
          </PrivateRoute>
          <PrivateRoute
              path="/account"
              requiredRoles={['administrator']}>
            <AccountManagementLayout />
          </PrivateRoute>
          <Route path="/logout">
            <Logout />
          </Route>
          <PrivateRoute
              path="/planning"
              requiredRoles={['planner', 'administrator']}>
            <ManualPlanning />
          </PrivateRoute>
          <PrivateRoute
              path="/view"
              requiredRoles={['view-only', 'planner', 'administrator']}>
            <ViewPlanning />
          </PrivateRoute>
          <PrivateRoute
              path="/data"
              requiredRoles={['planner', 'administrator']}>
            <DataVisualization />
          </PrivateRoute>
          <PrivateRoute
              path="/monthly"
              requiredRoles={['view-only', 'planner', 'administrator']}
          >
            <MonthlyDataAnalytics/>
          </PrivateRoute>
          <PrivateRoute
              path="/" exact
              requiredRoles={['view-only', 'planner', 'administrator']}>
            <Home />
          </PrivateRoute>
          <Route>
            <Redirect to='/'/>
          </Route>
        </Switch>
    )
}