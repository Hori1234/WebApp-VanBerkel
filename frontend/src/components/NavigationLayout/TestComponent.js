import React, { Component } from "react";
import { Menu, Dropdown, Button } from "antd";
import "antd/dist/antd.css";

export default class TestComponent extends Component {
  constructor(props) {
    super(props);
  }

  render() {
    const menu = (
      <Menu>
        <Menu.Item>
          <a onClick={this.props.UserVO}>View Only</a>
        </Menu.Item>
        <Menu.Item>
          <a onClick={this.props.UserAdmin}>Administrator</a>
        </Menu.Item>
        <Menu.Item>
          <a onClick={this.props.UserPlanner}>Planner</a>
        </Menu.Item>
      </Menu>
    );
    return (
      <div id="user-types">
        <Dropdown overlay={menu} placement="bottomLeft" arrow>
          <Button>User Types</Button>
        </Dropdown>
      </div>
    );
  }
}
