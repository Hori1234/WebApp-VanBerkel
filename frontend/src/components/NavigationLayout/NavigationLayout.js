import React, { Component , useState} from 'react'
import 'antd/dist/antd.css';
import { 
    Layout, 
    Menu, 
    Avatar,
    Divider , } from 'antd';
import { 
    UserOutlined, 
    LaptopOutlined,
     
    NotificationOutlined 
} from '@ant-design/icons';

import ButtonComponent from "../ButtonComponent"
import SignInComponent from "../SignIn/SignInComponent"
import TestComponent from "./TestComponent"

import "../Css/NavigationLayout.css"
const { SubMenu } = Menu;
const { Header, Content, Sider, Footer } = Layout;

export default class NavigationLayout extends Component {
    
    constructor(props) {
        super(props)
        this.state = { 
            isEmptyState: true ,
            isMenuItemClickedState: false,
            UserState: false,
        }
    };

    triggerAddTripState = (value) => {
        this.setState({
          ...this.state,
          isMenuItemClickedState: value,
        })
    };

    changeUser = (value) => {
        this.setState({
            ...this.state,
            UserState: value,
        })
        console.log(this.state.UserState);
    };
    
    render() {
        
        
        return (
            <Layout style={{ height: "100vh" }}>
                
                <Layout style={{margin: 10}}>
                    
                    <Sider width={200} className="site-layout-background">
                        <Layout 
                            className="dividers-style" >
                            <Avatar
                                style={{marginTop: 10}}
                                shape="square" 
                                size={64} 
                                icon={<UserOutlined />} />
                            <Divider>T.R.U.C.K.</Divider>
                           <TestComponent 
                            UserVO={() => this.changeUser("View Only")} 
                            UserAdmin={() => this.changeUser("Administrator")}
                            UserPlanner={() => this.changeUser("Planner")}
                            />
                        </Layout>
                        <Menu
                            mode="inline"
                            defaultSelectedKeys={['1']}
                            defaultOpenKeys={['sub1']}
                            style={{ height: '100%', borderRight: 0 }}
                        >
                           {this.state.UserState == 'Administrator' &&
                            
                                <SubMenu key="sub1" icon={<UserOutlined />} title="User Type: ">
                                    <Menu.Item key="1" onClick={() => this.triggerAddTripState('am')}>Account Management</Menu.Item>
                                    <Menu.Item key="2" onClick={() => this.triggerAddTripState('upp')}>Upload</Menu.Item>
                                    <Menu.Item key="3" onClick={() => this.triggerAddTripState('cp')}>Create Planning</Menu.Item>
                                    <Menu.Item key="4" onClick={() => this.triggerAddTripState('vp')}>View Planning</Menu.Item>
                                    <Menu.Item key="5" onClick={() => this.triggerAddTripState('dv')}>Data Visualization</Menu.Item>
                                    <Menu.Item key="6" onClick={() => this.triggerAddTripState('mda')}>Monthly Data Analytics</Menu.Item>
                                    
                                </SubMenu>
                            }
                            
                            {this.state.UserState == 'Planner' &&
                            
                                <SubMenu key="sub1" icon={<UserOutlined />} title="User Type: ">
                                    <Menu.Item key="2" onClick={() => this.triggerAddTripState('upp')}>Upload</Menu.Item>
                                    <Menu.Item key="3" onClick={() => this.triggerAddTripState('cp')}>Create Planning</Menu.Item>
                                    <Menu.Item key="4" onClick={() => this.triggerAddTripState('vp')}>View Planning</Menu.Item>
                                    <Menu.Item key="5" onClick={() => this.triggerAddTripState('dv')}>Data Visualization</Menu.Item>
                                    <Menu.Item key="6" onClick={() => this.triggerAddTripState('mda')}>Monthly Data Analytics</Menu.Item>
                                </SubMenu>
                            }

                            {this.state.UserState == 'View Only' &&
                            
                                <SubMenu key="sub1" icon={<UserOutlined />} title="User Type: ">
                                    <Menu.Item key="4" onClick={() => this.triggerAddTripState('vp')}>View Planning</Menu.Item>
                                    <Menu.Item key="5" onClick={() => this.triggerAddTripState('dv')}>Data Visualization</Menu.Item>
                                    <Menu.Item key="6" onClick={() => this.triggerAddTripState('mda')}>Monthly Data Analytics</Menu.Item>
                                    
                                </SubMenu>
                            }

                        </Menu>
                    </Sider>
                    <Layout style={{ padding: '0 24px 24px' , height: '100%', }}>
                        <Layout  
                            style={{ 
                                    marginTop: 25,
                                    minHeight: 280,
                                    alignItems:'center',
                                    width: '100%', }}>
                            <Content
                                className="site-layout-background"
                                style={{
                                    padding: 24,
                                    marginTop: 25,
                                    minHeight: 280,
                                    width: '50%'
                                }}
                                >
                                <Layout 
                                    style={{
                                        padding: 24,
                                        width: '100%',
                                        height: '100%',
                                        alignItems: "center"
                                    }}>
                                    {this.state.isMenuItemClickedState == "login" && <SignInComponent/> }

                                    {this.state.isMenuItemClickedState == "usracc" && <ButtonComponent />}
                                    
                                </Layout>
                            
                            </Content>
                        </Layout>
                        <Footer style={{ textAlign: 'center' }}>T.R.U.C.K. Â©2020 Created by SEP Group 2</Footer>
                    </Layout>
                    
                </Layout>

            </Layout>
        )
    }
}
