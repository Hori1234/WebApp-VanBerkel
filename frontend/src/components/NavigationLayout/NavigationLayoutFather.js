import { BrowserRouter } from "react-router-dom";
import NavigationLayout from "./NavigationLayout";
import React from "react";
import "antd/dist/antd.css";
import "../Css/NavigationLayout.css";

export default function NavigationLayoutFather(){
    return(
        <BrowserRouter><NavigationLayout/></BrowserRouter>
    )        
}

