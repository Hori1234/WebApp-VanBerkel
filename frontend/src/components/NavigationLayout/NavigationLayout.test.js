import React, { useContext } from 'react';
import { mount, shallow, render, setProps } from 'enzyme';
import {
    MemoryRouter
} from 'react-router'
import { Route, Link, useLocation } from 'react-router-dom';
import { Layout, Image, Typography, Button, Divider } from 'antd';
import AuthContext from '../contextConfig';
import NavigationLayout from './NavigationLayout';




describe('NAvigationLayout component', () => {


    it('should render the login page correctly', () => {

        const component = shallow(<NavigationLayout />);
        expect(component).toMatchSnapshot();
    });

    it('should have the correct view button href', () => {
        const user = { state: { username: 'test', role: 'Administrator' } };

        const component = render(<AuthContext.Provider value={user} >
            <NavigationLayout />
        </AuthContext.Provider>
        );
        const viewButton = component.find('#viewButton > a')[0];

        expect(viewButton.attribs.href).toEqual('/view');

    });

    it('should have the correct planningButton href', () => {
        const user = { state: { username: 'test', role: 'Administrator' } };

        const component = render(<AuthContext.Provider value={user} >
            <NavigationLayout />
        </AuthContext.Provider>
        );

        const dataButton = component.find('#dataButton > a')[0];

        expect(dataButton.attribs.href).toEqual('/data');

    });

    it('should have the correct planningButton href', () => {
        const user = { state: { username: 'test', role: 'Administrator' } };

        const component = render(<AuthContext.Provider value={user} >
            <NavigationLayout />
        </AuthContext.Provider>
        );

        const monthlyButton = component.find('#monthlyButton > a')[0];

        expect(monthlyButton.attribs.href).toEqual('/montly');

    });

    it('should have the correct uploadButton href', () => {
        const user = { state: { username: 'test', role: 'Administrator' } };

        const component = render(<AuthContext.Provider value={user} >
            <NavigationLayout />
        </AuthContext.Provider>
        );

        const uploadButton = component.find('#uploadButton > a')[0];

        expect(uploadButton.attribs.href).toEqual('/upload');

    });



});