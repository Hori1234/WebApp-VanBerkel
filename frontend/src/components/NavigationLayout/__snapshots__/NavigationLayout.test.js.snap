// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NAvigationLayout component should render the login page correctly 1`] = `
<BrowserRouter>
  <Layout
    style={
      Object {
        "height": "100vh",
      }
    }
  >
    <Layout
      style={
        Object {
          "margin": 10,
        }
      }
    >
      <Sider
        className="site-layout-background"
        collapsed={false}
        collapsible={true}
        onCollapse={[Function]}
        width={200}
      >
        <Layout
          className="dividers-style"
        >
          <Avatar
            icon={<ForwardRef(UserOutlined) />}
            shape="square"
            size={64}
            style={
              Object {
                "marginTop": 10,
              }
            }
          />
          <Divider>
            T.R.U.C.K.
          </Divider>
        </Layout>
        <Menu
          defaultOpenKeys={
            Array [
              "sub1",
            ]
          }
          mode="inline"
          style={
            Object {
              "borderRight": 0,
              "height": "100%",
            }
          }
        />
      </Sider>
      <Layout
        style={
          Object {
            "height": "100%",
            "padding": "0 24px 24px",
          }
        }
      >
        <Layout
          style={
            Object {
              "alignItems": "center",
              "marginTop": 25,
              "minHeight": 280,
              "width": "100%",
            }
          }
        >
          <Content
            className="site-layout-background"
            style={
              Object {
                "marginTop": 25,
                "minHeight": 280,
                "padding": 24,
                "width": "100%",
              }
            }
          >
            <Layout
              style={
                Object {
                  "alignItems": "center",
                  "backgroundColor": "white",
                  "height": "100%",
                  "padding": 24,
                  "width": "100%",
                }
              }
            >
              <Switch>
                <Route
                  path="/upload"
                >
                  <UploadButton />
                </Route>
                <Route
                  path="/account"
                >
                  <AccountManagementLayout />
                </Route>
                <Route
                  path="/planning"
                />
                <Route
                  path="/view"
                />
                <Route
                  path="/data"
                />
                <Route
                  path="/montly"
                />
                <Route
                  exact={true}
                  path="/"
                >
                  <Home />
                </Route>
              </Switch>
            </Layout>
          </Content>
        </Layout>
        <Footer
          style={
            Object {
              "textAlign": "center",
            }
          }
        >
          T.R.U.C.K. Â©2020 Created by SEP Group 2
        </Footer>
      </Layout>
    </Layout>
  </Layout>
</BrowserRouter>
`;
